megamek/src/megamek/common/battlevalue/BattleArmorBVCalculator.java

```diff
@@ -35,12 +49,14 @@
 
 public class BattleArmorBVCalculator extends BVCalculator {
 
-    private final BattleArmor battleArmor;
     private int currentTrooper;
 
     BattleArmorBVCalculator(Entity entity) {
         super(entity);
-        battleArmor = (BattleArmor) entity;
+    }
+
+    private BattleArmor getBattleArmor() {
```

Comment:
NGL, this is pretty gross.  I'd argue that _ideally_ it should either be getEntity() and the caller is responsible for casting, or getBattleArmor() but the class is instantiated with a BattleArmor instance, not an Entity instance.
I guess that would require rewriting the whole BVCalculator hierarchy, though...

Reply:
PS.: I have a possible solution for all those stateful calculators that are reutilized while not reutilizing anything and only making the code prone to ConcurrentModificationExceptions just because someone changed screens too fast.